apiserver/application/application_test.go

```diff
@@ -1094,7 +1094,77 @@ func (s *serviceSuite) TestServiceDeployConfigError(c *gc.C) {
 	c.Assert(err, jc.Satisfies, errors.IsNotFound)
 }
 
-func (s *serviceSuite) TestServiceDeployToMachine(c *gc.C) {
+func (s *serviceSuite) TestApplicationDeployConstraints(c *gc.C) {
```

Comment:
Please also include some idea of expected outcome in the name of your tests. Style is up to you, but I prefer this: `TestAppDeployConstraints_PicksCorrectType`